index.bs

```diff
@@ -681,14 +706,551 @@ id=] must be the same as the [=window handle=].
+
+      <dt>|value| is a [=platform object=] that implements [=WindowProxy=]
+      <dd>1. Let |remote value| be an [=object=] matching the <code>WindowProxyValue</code>
+             production in the [=local end definition=], with the <code>objectId</code>
+             property set to the [=object id for an object=] |value|.
+
+      <dt>|value| is a [=platform object=]
+      <dd>1. Let |remote value| be an [=object=] matching the <code>ObjectValue</code>
+             production in the [=local end definition=], with the <code>objectId</code>
+             property set to the [=object id for an object=] |value|.
+
+      <dt>[=IsCallable=](|value|)
+      <dd>Let |remote value| be an [=object=] matching the <code>FunctionValue</code>
+          production in the [=local end definition=], with the <code>objectId</code>
+          property set to the [=object id for an object=] |value|.
+
+      <dt>Otherwise:
+      <dd>
+      1. let |serialized| be null.
+
+      1. If |value| is not in the |set of known objects|, and |max depth|
+         is not null and greater than 0, run the following steps:
+         1. Append |value| to the |set of known objects|
+
+         1. Let |serialized| be the result of [=serialize as a mapping=] given
+            [=EnumerableOwnPropertyNames=](|value|, key+value), |max depth|, |node
+            details| and |set of known objects|
+
+      1. Let |remote value| be an [=object=] matching the <code>ObjectValue</code> production
+         in the [=local end definition=], with the <code>objectId</code> property set
+         to the [=object id for an object=] |value|, and the <code>value</code> field
+         set to |serialized|.
+    </dl>
+
+  1. Return |remote value|
+
+Issue: the way we check types is all wrong
+
+Issue: Does it make sense to use the same depth parameter for nodes and objects
+in general?
+
+</div>
+
+<div algorithm>
+Issue: this assumes for-in works on iterators
+
+To <dfn>serialize as a list</dfn> given |iterable|, |max depth|,
+|node details| and |set of known objects|:
+
+  1. Let |serialized| be a new list.
+
+  1. For each |child value| in |iterable|:
+
+    1. Let |serialized child| be the result of [=serialize as a
+       remote value=] with arguments |child value|, |max
+       depth| - 1, |node details| and |set of known objects|.
+
+    1. Append |serialized child| to |serialized|.
+
+  1. Return [=CreateArrayFromList=](|serialized|)
+</div>
+
+<div algorithm>
+To <dfn>serialize as a mapping</dfn> given |iterable|, |max depth|,
+|node details| and |set of known objects|:
+
+  1. Let |serialized| be a new list.
+
+  1. For |item| in |iterable|:
+    1. Assert: [=IsArray=](|item|)
+
+    1. Let |property| be [=CreateListFromArrayLike=](|item|)
+
+    1. Assert |property| is a list of size 2
+
+    1. Let |key| be |property|[0] and let |value| be |property|[1]
+
+    1. If [=Type=](|key|) is String, let |serialized key| be |child key|,
```

Comment:
Could *key* be anything other than a string here?

Reply:
It can be Symbol.

Reply:
I think it currently can't, but I'll resolve this in favor of the discussion in https://github.com/w3c/webdriver-bidi/pull/57#discussion_r508596239.